File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitDefects/EngineerVisitDefects.cs
using System.Collections;

namespace FSM.Entity
{
    namespace EngineerVisitDefectss
    {
        public class EngineerVisitDefects
        {
            private DataTypeValidator _dataTypeValidator;

            public EngineerVisitDefects()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _EngineerVisitDefectID = 0;

            public int EngineerVisitDefectID
            {
                get
                {
                    return _EngineerVisitDefectID;
                }
            }

            public object SetEngineerVisitDefectID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerVisitDefectID", value);
                }
            }

            private int _CategoryID = 0;

            public int CategoryID
            {
                get
                {
                    return _CategoryID;
                }
            }

            public object SetCategoryID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_CategoryID", value);
                }
            }

            private string _Reason = string.Empty;

            public string Reason
            {
                get
                {
                    return _Reason;
                }
            }

            public object SetReason
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Reason", value);
                }
            }

            private string _ActionTaken = string.Empty;

            public string ActionTaken
            {
                get
                {
                    return _ActionTaken;
                }
            }

            public object SetActionTaken
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ActionTaken", value);
                }
            }

            private bool _WarningNoticeIssued = false;

            public bool WarningNoticeIssued
            {
                get
                {
                    return _WarningNoticeIssued;
                }
            }

            public object SetWarningNoticeIssued
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_WarningNoticeIssued", value);
                }
            }

            private bool _Disconnected = false;

            public bool Disconnected
            {
                get
                {
                    return _Disconnected;
                }
            }

            public object SetDisconnected
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Disconnected", value);
                }
            }

            private string _Comments = string.Empty;

            public string Comments
            {
                get
                {
                    return _Comments;
                }
            }

            public object SetComments
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Comments", value);
                }
            }

            private int _AssetID = 0;

            public int AssetID
            {
                get
                {
                    return _AssetID;
                }
            }

            public object SetAssetID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AssetID", value);
                }
            }

            private int _EngineerVisitID = 0;

            public int EngineerVisitID
            {
                get
                {
                    return _EngineerVisitID;
                }
            }

            public object SetEngineerVisitID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerVisitID", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitDefects/EngineerVisitDefects.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace EngineerVisitDefectss

Public Class EngineerVisitDefects

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "EngineerVisitDefects Properties"

      
Private _EngineerVisitDefectID As Integer = 0
Public Readonly Property EngineerVisitDefectID() As Integer
	Get 
		Return _EngineerVisitDefectID
	End Get	
End Property	
Public Writeonly Property SetEngineerVisitDefectID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_EngineerVisitDefectID", Value)
	End Set
End Property


Private _CategoryID As Integer = 0
Public Readonly Property CategoryID() As Integer
	Get 
		Return _CategoryID
	End Get	
End Property	
Public Writeonly Property SetCategoryID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_CategoryID", Value)
	End Set
End Property


Private _Reason As String = String.Empty
Public Readonly Property Reason() As String
	Get 
		Return _Reason
	End Get	
End Property	
Public Writeonly Property SetReason() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Reason", Value)
	End Set
End Property


Private _ActionTaken As String = String.Empty
Public Readonly Property ActionTaken() As String
	Get 
		Return _ActionTaken
	End Get	
End Property	
Public Writeonly Property SetActionTaken() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_ActionTaken", Value)
	End Set
End Property


Private _WarningNoticeIssued As Boolean = False
Public Readonly Property WarningNoticeIssued() As Boolean
	Get 
		Return _WarningNoticeIssued
	End Get	
End Property	
Public Writeonly Property SetWarningNoticeIssued() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_WarningNoticeIssued", Value)
	End Set
End Property


Private _Disconnected As Boolean = False
Public Readonly Property Disconnected() As Boolean
	Get 
		Return _Disconnected
	End Get	
End Property	
Public Writeonly Property SetDisconnected() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Disconnected", Value)
	End Set
End Property


Private _Comments As String = String.Empty
Public Readonly Property Comments() As String
	Get 
		Return _Comments
	End Get	
End Property	
Public Writeonly Property SetComments() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Comments", Value)
	End Set
End Property


Private _AssetID As Integer = 0
Public Readonly Property AssetID() As Integer
	Get 
		Return _AssetID
	End Get	
End Property	
Public Writeonly Property SetAssetID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_AssetID", Value)
	End Set
End Property


Private _EngineerVisitID As Integer = 0
Public Readonly Property EngineerVisitID() As Integer
	Get 
		Return _EngineerVisitID
	End Get	
End Property	
Public Writeonly Property SetEngineerVisitID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_EngineerVisitID", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

